draft-ietf-sedate-datetime-extended.md

```diff
@@ -357,6 +357,57 @@ not specifically configured to take part in such an experiment.
 See {{BCP178}} for a discussion about the danger of experimental keys
 leaking out to general production and why that MUST be prevented.
 
+## Optionally Critical
+
+For the format defined here, suffix tags are always *optional*: They
+can be added or left out as desired by the generator of the string in
+Internet Date/Time Format.  (An application might require the presence
+of specific suffix tags, though.)
+
+Without further indication, they are also *elective*: Even if included
+in the Internet Date/Time Format string, the recipient is free to
+ignore the suffix tag.  Reasons might include that the recipient does
+not implement (or know about) the specific suffix key, or that it does
+recognize the key but cannot act on the value provided.
+
+A suffix tag may also indicate that it is *critical*: The recipient is
+advised that it MUST not act on the Internet Date/Time Format string
+unless it can process the suffix tag as specified.  A critical suffix
+tag is indicated by following its opening bracket with an exclamation
+mark (see `critical-flag` in {{abnf}}).
+
+A string such as:
+
+    2022-07-08T00:14:07Z[Europe/Paris]
+
+is internally inconsistent, as Europe/Paris does not use a time zone
+offset of 0.
+The time zone hint given is elective, though, so the recipient is not
+required to act on the inconsistency; it can treat the Internet
+Date/Time Format string as if it were:
+
+    2022-07-08T00:14:07Z
+
+Similar with:
+
+    2022-07-08T00:14:07Z[knort=blargel]
+
+However,
+
+    2022-07-08T00:14:07Z[!Europe/Paris]
```

Comment:
Do you have a proposal for a critical-flag that is forward-compatible with JAVAZDT?

Reply:
Nope, the Java parser is very picky (e.g. won't even accept `-00:00`) so any change to the string format won't be accepted by JAVAZDT.  (Although of course a later version of Java's standard library could support it.)

I'm mostly wondering if it's worth adding a note warning clients that the critical flag is new syntax that will be incompatible with existing users of the time zone suffix like JAVAZDT.  If this kind of warning isn't usually done in IETF specs, then IMO it's OK to omit.

Reply:
I agree with @cabo here. This is new syntax, but the old syntax is also supported: this just adds an additional capability that might be easily added in a future version of JAVAZDT without breaking anything.

Reply:
> What about the [u-ca=...] feature in JAVAZDT?

Current Java won't accept any other suffixes either. But 99%+ usage in Temporal won't emit that calendar suffix either, because non-Gregorian calendars are a rarely-used feature outside of localized formatting.

Regardless, if we don't think a note is needed in this case then I'm OK with that. Resolving.

Reply:
Oh, looks like I can't resolve conversations I started. :-) Anyway, this one seems OK & should be resolved.